coupa-ops/aws
# CO-7978 - Keep last n snapshots regardless of how old they are

[CO-7978](https://coupadev.atlassian.net/browse/CO-7978) - Keep last n snapshots regardless of how old they are

- Check for the lineage_tag in data bag before restore
- Keep last n snapshots regardless of how old they are
- keep consistent snapshot lineage in AWS, Azure and cloudstack

## JIRA

* [Main JIRA ticket](https://coupadev.atlassian.net/browse/CO-7978)

## Code reviewers

- [ ] @coupacooke 
- [ ] @charchitcoupa 
- [x] @alokdnb 
- [ ] @Ramesh7 

## Summary of issue

- Unable to restore DLU snapshot without renaming the lineage
- Snapshots getting deleted as per the deletion logic once the lineage is renamed in order to restore the snapshot

## Summary of change

- Restore a DLU snapshot without renaming the lineage
- Update the data bag and set the `lineage_tag` as mentioned [here](https://coupadev.atlassian.net/wiki/spaces/CO/pages/594216114/Restore+volume+from+snapshot+with+timestamp) 

```json
"storage": {
    "prdxxxdb": {
      "timestamp": 1552131422,
      "lineage_tag": "prdxxxdb-controlupgrade"
    }
  }
```

## Testing approach

- Update the deployment data bag
- Set `timestamp`
- Set `lineage_tag`
- Relaunch the DBS
- Verify the snapshot id
